apps/system/flash_test/flash_test.c

```diff
@@ -84,18 +351,55 @@ static int flash_read(unsigned long addr, int size)
 	int remaining;
 	FAR uint8_t *buffer;
 	FAR struct mtd_dev_s *dev_mtd = NULL;
+	long long total_duration_ms = 0;
+	systick_time_ms = 0;
 
 	if (addr == 0 || size <= 0) {
 		printf("Invalid arguments, offset 0x%x, read size %d\n", addr, size);
 		return ERROR;
 	}
+	if (flash == FLASH_INTERNAL) {
+		printf("Received Flash Read cmd for internal flash offset 0x%x, read size %d\n", addr, size);
+		dev_mtd = up_flashinitialize();
+		if (!dev_mtd) {
+			printf("up_flashinitialize Failed\n");
+			return ERROR;
+		}
 
-	dev_mtd = up_flashinitialize();
-	if (!dev_mtd) {
-		printf("up_flashinitialize Failed\n");
-		return ERROR;
+	}
+#ifdef CONFIG_SECOND_FLASH_PARTITION
+
+	else if (flash == FLASH_EXT) {
+		printf("Received Flash Read cmd for external flash offset 0x%x, read size %d\n", addr, size);
+		struct spi_dev_s *spi = up_spiinitialize(1);
```

Comment:
Will this be callable using loadable defconfig? It should get a compile error.

Reply:
using the flash driver only for flat build update the app accordingly